pkg/virt-controller/watch/migration.go

```diff
@@ -1444,3 +1454,41 @@ func isNodeSuitableForHostModelMigration(node *k8sv1.Node, pod *k8sv1.Pod) bool
 
 	return true
 }
+
+func (c *MigrationController) matchMigrationPolicy(vmi *virtv1.VirtualMachineInstance, clusterMigrationConfiguration *virtv1.MigrationConfiguration) error {
```

Comment:
I wonder what's the advantage of matching the policy here in virt-controller and not in the virt-handler directly?

Reply:
Please see David's comment [here](https://github.com/kubevirt/kubevirt/pull/6399#pullrequestreview-814193598).

I think he has a valid point. It makes sense that `handleTargetPodHandoff()` will, as its name suggest, handle this logic and virt-handler will simply use the needed migration configurations that are defined.

Reply:
Another reason is that `vmUpdateHelperMigrationSource()` is not meant to modify the VMI in any way, while `handleTargetPodHandoff()` does with `patchVMI()`. This is important since we want to know while policy was matched for the migration, and we need to patch VMI status for that.

Reply:
virt-handler updates the VMI status in many places already, I'm not sure why would this be a concern.

My main concern is that at some point we will want to choose the policy also according to some parameters that are known to virt-handler. 
It may be for example a negotiation the virt-handler may do with the destination (like measuring the bandwidth or anything else that I cannot think of yet).
Implementing the matching logic in virt-controller will make us comminicate this information to virt-controller. 
I don't have a concrete usecase right, so I'm just raising this as a concern of mine.
@davidvossel fyi

Reply:
> My main concern is that at some point we will want to choose the policy also according to some parameters that are known to virt-handler.

> Implementing the matching logic in virt-controller will make us comminicate this information to virt-controller.
I don't have a concrete usecase right, so I'm just raising this as a concern of mine.

If we get to that point, I think we have a backwards compatible way of transitioning policy matching to virt-handler. We'd just make virt-controller stop setting the vmi.Status.MigrationState.MigrationConfiguration field and have virt-handler fill it in. The api wouldn't have to change.

Reply:
Yeah, I agree. Maybe we could also solve this with additional node labels.